### ISSUE TITLE
Windows: "Access is denied" error and backup stops

### ISSUE THREAD

**Issue by mholt** [restic/restic#1206]
## Output of `restic version`

restic 0.7.1 (v0.7.1-118-g0ee1650f)
compiled with go1.9 on windows/amd64

## How did you run restic exactly?

```
restic backup --exclude "_not_backed_up" --hostname "MyPC" D:\
```

## What backend/server/service did you use?

Local disk

## Expected behavior

Restic should log any errors accessing files it discovers, but continue the backup anyway

## Actual behavior

Restic stopped at 3.03% with:

```
error for \\?\D:\System Volume Information: open \\?\D:\System Volume Information: Access is denied.

```

CPU and disk activity went to basically 0.

I hit Ctrl+C and it got to 3.08% and got stuck again, but the error message didn't change. Then I hit Ctrl+C *again* and it is still going, now at about 6%. I won't be here much longer to see it finish but I hope it will finish!

## Steps to reproduce the behavior

No idea, try backing up a drive with "System Volume Information" I guess?

## Do you have any idea what may have caused this?

I ran restic without being Administrator.

## Do you have an idea how to solve the issue?

See "Expected behavior" above

---

I'm not sure if I just misconfigured restic, but I have this backup scheduled weekly, it'd be nice if it just skipped over permissions errors (with an optional log). However, if I've misconfigured something, do let me know. :)

**Comment by mholt** [comment]
Update: Okay, at about 11% it got stuck again and I had to give it another kick with Ctrl+C. (The error message didn't change.)

**Comment by fd0** [comment]
Hm, this is odd indeed. It should print an error message and then continue. So it looks like you've stumbled upon a bug.

Where is the repository located? What was your current working directory when running restic?

**Comment by mholt** [comment]
The repository is located in E:\restic_backup. The current directory when I ran this particular command was, I think, just %HOME%.

**Comment by fd0** [comment]
Thanks for the details. I need to find a Windows machine and have a look for myself. I think this may be a bug that only occurs on Windows. Hm.

**Comment by mholt** [comment]
I'd help, but I don't have Windows either. (This is on a family member's computer...)

**Comment by mholt** [comment]
Incidentally, what is Ctrl+C (SIGINT) supposed to do? I was kind of surprised it didn't kill the whole process, but somewhat pleased, I guess, that the "kick" got it un-stuck.

**Comment by fd0** [comment]
Good question. I'm not really sure what happened here. I'm surprised that it helped to move the process forward in your case.

On Unix, restic handles SIGINT by removing the locks it left in the repo, then it exits. This has the (unintended) side effect that when a remote repo is used, it will properly finish uploading first before the lock can be removed.

I have plans to properly tear down the internal processing pipelines (using `context`), but that's not in place yet.

**Comment by mholt** [comment]
Ah, one thing to try would be these free Windows VMs offered by Microsoft: https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/

I'm afraid I won't have time to do this in a while though, as classes are starting :(

**Comment by mholt** [comment]
Well, I've had this running as a scheduled task every week on my parents' computer and the task does seem to stop after a while. I need to investigate if that's because the scheduler is killing it after a while or if it's because, under the hood, restic is actually continuing and finishing. I probably should observe it and check CPU and disk usage... I thought I did this but let me do it again and confirm.

**Comment by fd0** [comment]
Would you mind checking if this is still an issue with 0.9.0?

**Comment by mholt** [comment]
I have good news.

I just went to my parents' house today and tried it out, but after I downloaded 0.9.0 and ran `restic snapshots`, I found that, indeed, snapshots had been created every week for the last year! (I had set up a task in the Scheduler to run it regularly.)

So, I guess that means the backups are completing. Now it is running 0.9 so I will check up with it again in a few weeks.

Thanks for the ping!
